### ISSUE TITLE
Evaluation results on several models are lower than reported ones.

### ISSUE THREAD

**Issue by xioryu** [zhreshold/mxnet-ssd#131]
I simply downloaded the released ssd models pretrained on voc07+12. The models are evaluated with the latest mxnet ssd examples. The evaluation results are:
Model                                       Test data                 mAP (evaluated)        mAP (reported)
VGG16_reduced 300x300        VOC07 test              75.7                             77.8
VGG16_reduced 512x512        VOC07 test                                                  79.9
Inception-v3 512x512             VOC07 test               77.1                             78.9
Resnet-50 512x512                  VOC07 test              77.8                             78.9

Evaluation is performed with symbols legacy_vgg16_ssd_300, legacy_vgg16_ssd_512, inceptionv3 and resnet50. There are 1~2 mAP drops. 

I also tried reproducing the ssd_vgg_reduced_300 model and got 73.1~73.7 mAP after several experiments. Could someone offer me any suggestions on how to reproduce them ?

**Comment by xioryu** [comment]
@zhreshold

**Comment by zhreshold** [comment]
@xioryu 
Do you have time writing out the results to files and use official Matlab code to verify the results?

**Comment by davsol** [comment]
Adding to this list, the new mobileNet512x512 gives 67.5 mAP

**Comment by jyuan1118** [comment]
@davsol Not really. The mAP of MobileNet 512x512 is 72.5 on VOC07 test. MobileNet 608x608 gives 74.57. I evaluated this last week.

**Comment by davsol** [comment]
You are right, I discovered a bug in my code and when I fixed it, the
results were the same as reported

**Comment by cc-cn** [comment]
@zhreshold ,  Ｉ have almost the same results with @xioryu

**Comment by xioryu** [comment]
@davsol Could you tell me which kind of bug in the code?

**Comment by davsol** [comment]
I automated the scales of the anchors, and my calculations were wrong. This
resulted with a model that learned on certain scales, and was evaluated on
others.
 Working on a new generic function now :)

**Comment by cypw** [comment]
Hi @xioryu , I also cannot reproduce the reported accuracy.

I doubt if the reported results are reliable since I noticed that the reported accuracy is posted before the evaluation metric fixed ( issuse #97 ). Do you think this the reason why we cannot reproduce the results?

Note:
The VGG's results are posted before [Apr 4, 2017](https://github.com/zhreshold/mxnet-ssd/commit/67ba1c9a8cb0508b018e958fd540f1e99423d0e3#diff-04c6e90faac2675aa89e2176d2eec7d8)
The evaluation metric is fixed on [Jun 29, 2017](https://github.com/zhreshold/mxnet-ssd/commit/ee116968edff60db7cf12cbff8182996ab5278b2)

**Comment by zhreshold** [comment]
I am going to retrain some of the models to be consistent with recent updates. Don't worry.

**Comment by zhreshold** [comment]
Guys, I've uploaded a new resnet50, see if it works for you. Next one is inceptionv3. 
BTW, the updated vgg model is still not good enough. I will revisit back to it later.

**Comment by xioryu** [comment]
The new resnet50 works. I got 79.01 mAP (similar as reported)

**Comment by BunnyShan** [comment]
@zhreshold How did you get this? Could you please share your configuration?

**Comment by zhreshold** [comment]
@BunnyShan It's basically the standard arguments.
```
python train.py --gpus 0,1,2,3 --network resnet50 --lr 0.004 --freeze '' --data-shape 512
```

**Comment by BunnyShan** [comment]
@zhreshold Thanks, I will try it!

**Comment by BunnyShan** [comment]
@zhreshold I tried your config on resnet50 512 shape and got 79.5%mAP! But I can't reproduce 77.1% on resnet50 321 shape as reported in the paper, I also tried predict_module in DSSD, still could not reproduce the result. Any advice?

**Comment by zhreshold** [comment]
@BunnyShan I have some idea to tune the configs for models with smaller input size, will need to verify  that.

**Comment by BunnyShan** [comment]
@zhreshold I tried resnet101 that was used in the origin paper and got very poor result, need to notice that the default resnet101 layer config is different from origin paper but both cannot reproduce the result. I thought maybe the high level feature is not suitable for detection and I'm diving in the caffe implementation for some results. And I'm looking forward to your result on this issue :)

**Comment by zhreshold** [comment]
@BunnyShan Fixed https://github.com/zhreshold/mxnet-ssd/commit/e7cc662f3d0ce29e10670f60a8cfcce854036780

**Comment by BunnyShan** [comment]
@zhreshold Thanks a lot! Any progress on small input size?

**Comment by yahyanik** [comment]
@xioryu @zhreshold Can you kindly share the initial values used for training? I want to train the MobileNet 300 on VOC07 but the mAP is not near these numbers. Right now I am using the default values for the VGG-reduced.

**Comment by siyiding1216** [comment]
@zhreshold Are you going to retrain the VGG16 based SSD? We could not reproduce the precision using current training config.

**Comment by nopattern** [comment]
@zhreshold 
      I have checked the caffe ssd code , The low map may be caused by data augment.  The original caffe has the following order:
        (1) Image distortion;
       (2) Image expansion;
       (3) Sampling a random crop window;
       (4) Data transform: Adding noise, resizing, 
        The second step seems to be  ignored in this mxnet version ,

**Comment by zhreshold** [comment]
@nopattern mAP is no longer the issue in https://github.com/dmlc/gluon-cv 
I admit that data aug is quite tricky and prune to change, that's why I am not developing using C++ iterator anymore and transit to python in gluon.

**Comment by nopattern** [comment]
get it.
